pkg/osbuild/systemd_unit_create_stage.go

```diff
@@ -177,3 +229,112 @@ func NewSystemdUnitCreateStage(options *SystemdUnitCreateStageOptions) *Stage {
 		Options: options,
 	}
 }
+
+// GenSystemdMountStages generates a collection of
+// org.osbuild.systemd.unit.create stages with options to create systemd mount
+// units, one for each mountpoint in the partition table.
+func GenSystemdMountStages(pt *disk.PartitionTable) ([]*Stage, error) {
+	mountStages := make([]*Stage, 0)
+	unitNames := make([]string, 0)
+
+	genOption := func(ent disk.FSTabEntity, path []disk.Entity) error {
+		fsSpec := ent.GetFSSpec()
+		fsOptions, err := ent.GetFSTabOptions()
+		if err != nil {
+			return err
+		}
+
+		var filename string
+		var options *SystemdUnitCreateStageOptions
+		device := fmt.Sprintf("/dev/disk/by-uuid/%s", strings.ToLower(fsSpec.UUID))
+		if err := uuid.Validate(fsSpec.UUID); err != nil {
+			// vfat IDs and other non-UUID identifiers aren't lowercased
+			device = fmt.Sprintf("/dev/disk/by-uuid/%s", fsSpec.UUID)
+		}
+		switch ent.GetFSType() {
+		case "swap":
+			filename = fmt.Sprintf("%s.swap", pathEscape(device))
+			options = &SystemdUnitCreateStageOptions{
```

Comment:
Can we move options up to above the switch and then only assign the `Config.Swap` and `Config.Mount` properties in the switch itself? Otherwise they're quite similar.